src/Discord.Net.Core/CDN.cs

```diff
@@ -51,11 +51,18 @@ public static string GetDefaultUserAvatarUrl(ushort discriminator)
         /// </summary>
         /// <param name="guildId">The guild snowflake identifier.</param>
         /// <param name="iconId">The icon identifier.</param>
+        /// <param name="size">The size of the image to return in. This can be any power of two between 16 and 2048.</param>
+        /// <param name="format">The format to return.</param>
         /// <returns>
-        ///     A URL pointing to the guild's icon.
+        ///     A URL pointing to the guild's icon in the specified size.
         /// </returns>
-        public static string GetGuildIconUrl(ulong guildId, string iconId)
-            => iconId != null ? $"{DiscordConfig.CDNUrl}icons/{guildId}/{iconId}.jpg" : null;
+        public static string GetGuildIconUrl(ulong guildId, string iconId, ushort size, ImageFormat format)
+        {
```

Comment:
We should check that size is a power of two ([this looks like a good solution to do so](https://stackoverflow.com/a/600306)), and that it's in the range [16-2048].

Reply:
This *may* be better represented as an enum:
```cs
public enum ImageSize : ushort
{
    Sixteen = 16,
    ThirtyTwo = 32,
    SixtyFour = 64,
    OneTwentyEight = 128,
    TwoFiftySix = 256,
    FiveTwelve = 512,
    TenTwentyFour = 1024,
    TwentyFourtyEight = 2048
}
```

Reply:
I'm waiting on more opinions on this. To me, joe's enum solution seems a bit better. Anyways if such change would have been done, we would have to change the arguments we take for both the user's avatar url and the guild's icon url.